src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -107,7 +113,7 @@ object ScalafixPlugin extends AutoPlugin {
               compile.value // evaluated first, before the potential scalafix evaluation
             if (scalafixOnCompile.value)
               scalafix
-                .toTask("")
+                .toTask(scalafixArgsOnCompile.value.mkString(" ", " ", ""))
```

Comment:
it would be nice to update the scripted test about caching, to make sure that an explicit `scalafix` after a compile with `scalafixOnCompile` does run rules that are not defined in the `triggered` overlay.

Given the current cache implementation, I think an explicit `scalafix` invocation will no longer benefit from the cache of the previous `compile` invocations after this PR. That's safer, as we can't assume that the configuration is effectively the same and add `--triggered` in https://github.com/scalacenter/sbt-scalafix/blob/8284c35c07418bcf07095097ba7729a360ecf891/src/main/scala/scalafix/sbt/ScalafixPlugin.scala#L510-L511. I guess we could do so if we see that there is no `triggered` section in the configuration.

Reply:
Thank you for your suggestion. I'll check how cache works later!

Reply:
`--triggered` is considered as a cache key only if there is `triggered` section in `.scalafix.conf`.